tests/test_main.py

```diff
@@ -122,3 +124,24 @@ def test_run_match_config_params() -> None:
         if key not in ("app_dir",)
     }
     assert config_params == run_params
+
+
+@pytest.mark.anyio
+async def test_run_multiprocess_with_sockets():
+    config = Config(app=app, workers=2, limit_max_requests=1)
+    with socket.socket() as sock:
+        sock.bind(("localhost", 0))
+        async with run_server(config, sockets=[sock]) as server:
+            while not server.started:
+                await asyncio.sleep(0.1)
+            await asyncio.sleep(0.1)
+
+
+@pytest.mark.anyio
+async def test_run_invalid_host() -> None:
+    with pytest.raises(SystemExit) as e:
+        config = Config(app=app, host="illegal_host")
+        server = Server(config=config)
+        await server.serve()
+    assert e.type == SystemExit
+    assert e.value.code == 1
```

Comment:
I think we can assert something more meaningful with the `caplog` fixture... Can't we?
```suggestion
async def test_run_invalid_host(caplog: pytest.LogCaptureFixture) -> None:
    with pytest.raises(SystemExit):
        config = Config(app=app, host="illegal_host")
        server = Server(config=config)
        await server.serve()
```

Reply:
I will check this tonight